### ISSUE TITLE
"module 'salt.ext' has no attribute 'ipaddress

### ISSUE THREAD

**Issue by cmusser** [saltstack/salt#52559]
### Description of Issue/Question

We have a Salt module that uses the Python `ipaddress` module to get various system IP addresses. This has worked well until our most recent minion, where applying the state results in `module 'salt.ext' has no attribute 'ipaddress'` A similar issue was reported in https://github.com/saltstack/salt/issues/49588, which was reported as resolved.

### Setup

Not sure what setup is relevant, but can add as needed. Our master is Ubuntu 18.04.1 LTS and the minion that is failing is Ubuntu 18.04.2 LTS.

### Steps to Reproduce Issue
apply the state:

```
sudo salt ourhost.com state.apply
ourhost.com:
    Data failed to compile:
----------
    Rendering SLS 'base:sensu-client' failed: Jinja error: module 'salt.ext' has no attribute 'ipaddress'
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/salt/utils/templates.py", line 418, in render_jinja_tmpl
    output = template.render(**decoded_context)
  File "/usr/lib/python3/dist-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/lib/python3/dist-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python3/dist-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "<template>", line 22, in top-level template code
  File "/var/cache/salt/minion/extmods/modules/intf.py", line 8, in primary_addr4
    return __salt__['network.interface_ip'](primary_intf4())
  File "/var/cache/salt/minion/extmods/modules/intf.py", line 2, in primary_intf4
    return __salt__['network.default_route']('inet')[0]['interface']
  File "/usr/lib/python3/dist-packages/salt/modules/network.py", line 1648, in default_route
    _routes = routes()
  File "/usr/lib/python3/dist-packages/salt/modules/network.py", line 1605, in routes
    routes_ = _ip_route_linux()
  File "/usr/lib/python3/dist-packages/salt/modules/network.py", line 514, in _ip_route_linux
    address_mask = convert_cidr(comps[0])
  File "/usr/lib/python3/dist-packages/salt/modules/network.py", line 1138, in convert_cidr
    network_info = salt.ext.ipaddress.ip_network(cidr)
AttributeError: module 'salt.ext' has no attribute 'ipaddress'

; line 22

---
[...]
    - mode: 644
    - template: jinja
    - defaults:
        sensu_passwd: {{ pillar['sensu_client_passwd'] }}
        hostname: {{ grains['fqdn'] }}
        public_addr: {{ salt.intf.primary_addr4() }}    <======================
        subscriptions: {{ pillar['sensu_subscriptions'] }}
    - require:
      - pkg: sensu

/etc/sensu/ssl/cert.pem:
[...]
---
----------
    Rendering SLS 'base:sshd' failed: Jinja error: module 'salt.ext' has no attribute 'ipaddress'
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/salt/utils/templates.py", line 418, in render_jinja_tmpl
    output = template.render(**decoded_context)
  File "/usr/lib/python3/dist-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/lib/python3/dist-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python3/dist-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "<template>", line 9, in top-level template code
  File "/var/cache/salt/minion/extmods/modules/intf.py", line 8, in primary_addr4
    return __salt__['network.interface_ip'](primary_intf4())
  File "/var/cache/salt/minion/extmods/modules/intf.py", line 2, in primary_intf4
    return __salt__['network.default_route']('inet')[0]['interface']
  File "/usr/lib/python3/dist-packages/salt/modules/network.py", line 1648, in default_route
    _routes = routes()
  File "/usr/lib/python3/dist-packages/salt/modules/network.py", line 1605, in routes
    routes_ = _ip_route_linux()
  File "/usr/lib/python3/dist-packages/salt/modules/network.py", line 514, in _ip_route_linux
    address_mask = convert_cidr(comps[0])
  File "/usr/lib/python3/dist-packages/salt/modules/network.py", line 1138, in convert_cidr
    network_info = salt.ext.ipaddress.ip_network(cidr)
AttributeError: module 'salt.ext' has no attribute 'ipaddress'

; line 9

---
[...]
    - user: root
    - group: root
    - mode: 644
    - template: jinja
    - defaults:
        listen_addr: {{ salt.intf.primary_addr4() }}    <======================
        listen_port: {{ pillar['sshd_listen_port'] }}

sshd:
  service.running:
    - enable: True
[...]
---

```

### Versions Report
Minion:
```
salt-minion -V
Salt Version:
           Salt: 2017.7.4
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.6.7 (default, Oct 22 2018, 11:32:17)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.2.5
 
System Versions:
           dist: Ubuntu 18.04 bionic
         locale: UTF-8
        machine: x86_64
        release: 4.15.0-47-generic
         system: Linux
        version: Ubuntu 18.04 bionic
```

master:

```
Salt Version:
           Salt: 2018.3.3
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: 2.0.3
      gitpython: 2.1.8
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.7
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.15rc1 (default, Nov 12 2018, 14:31:15)
   python-gnupg: 0.4.1
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 2.0.3
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.2.5
 
System Versions:
           dist: Ubuntu 18.04 bionic
         locale: UTF-8
        machine: x86_64
        release: 4.15.0-1032-aws
         system: Linux
        version: Ubuntu 18.04 bionic
```

**Comment by s0undt3ch** [comment]
The fix was released in v2018.3.4, v2019.2.0 and will also be included in the upcoming v2017.7.9

**Comment by cmusser** [comment]
ok, I think I can arrange to use a more recent version.
